### PR TITLE
[zero] add chunk init function for users

### PR DESCRIPTION
**Pull request by 1SAA**

### CODE REVIEW

File: colossalai/gemini/chunk/search_utils.py [hunk 1]
```diff
@@ -1,100 +1,106 @@
...
+from typing import Dict, List, Tuple
+
+import numpy as np
+import torch.nn as nn
+
+from colossalai.tensor import ColoParameter
+
+
+def in_ddp(param: nn.Parameter) -> bool:
+    return not getattr(param, '_ddp_to_ignore', False)
```
Comment (feifeibear) [inline colossalai/gemini/chunk/search_utils.py:11]: what if the param has no attr named as _ddp_to_ignore

File: colossalai/gemini/chunk/search_utils.py [hunk 2]
```diff
@@ -1,100 +1,106 @@
...
+
+
+def _filter_exlarge_params(model: nn.Module, size_dict: Dict[int, List[int]]) -> None:
+    """Filter those parameters whose size is too large from others.
+    """
+    params_size = [p.numel() for p in model.parameters() if in_ddp(p)]
+    params_size_arr = np.array(params_size)
+
+    std = np.std(params_size_arr)
```
Comment (feifeibear) [inline colossalai/gemini/chunk/search_utils.py:20]: why not use torch.std? is it necessary to conver it to np?

File: colossalai/gemini/chunk/search_utils.py [hunk 3]
```diff
@@ -1,100 +1,106 @@
...
+
+from colossalai.tensor import ColoParameter
+
+
+def in_ddp(param: nn.Parameter) -> bool:
+    return not getattr(param, '_ddp_to_ignore', False)
+
+
+def _filter_exlarge_params(model: nn.Module, size_dict: Dict[int, List[int]]) -> None:
+    """Filter those parameters whose size is too large from others.
```
Comment (feifeibear) [inline colossalai/gemini/chunk/search_utils.py:15]: add more comments. How to define too large than others!

File: colossalai/gemini/chunk/search_utils.py [hunk 4]
```diff
@@ -1,100 +1,106 @@
...
+    std = np.std(params_size_arr)
+    mean = np.mean(params_size_arr)
+    upper_limit = mean + 3 * std
+
+    for key in size_dict:
+        org_list = size_dict[key]
+        size_dict[key] = list(filter(lambda x: x <= upper_limit, org_list))
+
+
+def _get_unused_byte(size_list: List[int], chunk_size: int) -> int:
```
Comment (feifeibear) [inline colossalai/gemini/chunk/search_utils.py:29]: are you sure the return value is in byte not element number?

File: colossalai/gemini/chunk/search_utils.py (hunk 1 above)
Comment (1SAA) [reply colossalai/gemini/chunk/search_utils.py:11]: returns a False by default.

File: colossalai/gemini/chunk/search_utils.py [hunk 5]
```diff
@@ -1,100 +1,106 @@
...
+    left = 0
+    for s in size_list:
+        if s > left:
+            acc += left
+            left = chunk_size
+        left -= s
+    return left + acc
+
+
+def clasify_params(model: nn.Module) -> Dict[int, List[ColoParameter]]:
```
[the remark this answers is not in the document]
Comment (feifeibear) [reply colossalai/gemini/chunk/search_utils.py:42]: please add comments or change a more straightforward name. Classify params w.r.t. their DP degree?

File: colossalai/gemini/chunk/search_utils.py (hunk 2 above)
Comment (1SAA) [reply colossalai/gemini/chunk/search_utils.py:20]: I don't know. @ver217 wrote this code.

File: colossalai/gemini/chunk/search_utils.py (hunk 4 above)
Comment (1SAA) [reply colossalai/gemini/chunk/search_utils.py:29]: 100% sure is not in byte. here byte is used to represent a minimum unit.

File: colossalai/gemini/chunk/utils.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+    config_dict, wasted_size = search_chunk_configuration(model, **kwargs_dict)
+
+    dist.barrier()
+    end = time()
+    span_s = end - begine
+    wasted_size /= 1024**2
+
+    if dist.get_rank() == 0:
+        print("searching chunk configuration is completed in {:.2f} s.\n".format(span_s),
```
[the remark this answers is not in the document]
Comment (1SAA) [reply colossalai/gemini/chunk/utils.py:50]: print_rank_0 is not robust enough to print multiple strings. it is only able to print a single string.

**Pull request merged**